runtime/gc_base/RootScanner.cpp

```diff
@@ -395,44 +395,44 @@ MM_RootScanner::scanPermanentClasses(MM_EnvironmentBase *env)
 {
 	reportScanningStarted(RootScannerEntity_PermanentClasses);
 
+	/* Do systemClassLoader */
+	scanClassloader(env, static_cast<J9JavaVM*>(_omrVM->_language_vm)->systemClassLoader);
+
+	/* Do applicationClassLoader */
+	scanClassloader(env, static_cast<J9JavaVM*>(_omrVM->_language_vm)->applicationClassLoader);
+
+	condYield();
+
+	reportScanningEnded(RootScannerEntity_PermanentClasses);
+}
+
+/**
+ * Scan all objects from class loader.
+ */
+void
+MM_RootScanner::scanClassloader(MM_EnvironmentBase *env, J9ClassLoader *classLoader)
+{
 	J9MemorySegment *segment = NULL;
 	J9Class *clazz = NULL;
 
-	/* Do systemClassLoader */
-	if (NULL != static_cast<J9JavaVM*>(_omrVM->_language_vm)->systemClassLoader) {
-		GC_ClassLoaderSegmentIterator segmentIterator(static_cast<J9JavaVM*>(_omrVM->_language_vm)->systemClassLoader, MEMORY_TYPE_RAM_CLASS);
-		while(NULL != (segment = segmentIterator.nextSegment())) {
-			if(_singleThread || J9MODRON_HANDLE_NEXT_WORK_UNIT(env)) {
+	if (NULL != classLoader) {
+		GC_ClassLoaderSegmentIterator segmentIterator(classLoader, MEMORY_TYPE_RAM_CLASS);
+		while (NULL != (segment = segmentIterator.nextSegment())) {
+			if (_singleThread || J9MODRON_HANDLE_NEXT_WORK_UNIT(env)) {
 				GC_ClassHeapIterator classHeapIterator(static_cast<J9JavaVM*>(_omrVM->_language_vm), segment);
```

Comment:
We have this pattern of fetching language VM from omr VM a number of times in RootScanner. We should probably just have a fieldd for  language VM instead for omr VM. Could be done in a separate change or just left as a first item for a new hire.